### PR TITLE
OCPBUGS-18305,RHIBMCS-151: Copied csv listing backport

### PR DESCRIPTION
**Pull request by dtfranz**
This is a selective backport of #532, which contains an optimization that appears to have fixed or reduced the frequency of stale CSVs in the catalog-operator cache. I have removed the changes to the olm operator as they require an `api` version bump which brings in new features.

### CODE REVIEW

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Comment by dinhxuanvu** [comment]
The OG CSV copy test failure is a bit iffy but I expect it to be a flake given this change didn't touch olm operator which handle OG/copied CSV logic. The other failure has nothing to do with copied CSV.
/retest

**Comment by awgreene** [comment]
/hold

I'd like to use https://issues.redhat.com/browse/OCPBUGS-5080 to backport this fix as this change was credited with solving https://issues.redhat.com/browse/OPRUN-3002, which is really just a duplicate of OCPBUGS-5080.

**Comment by awgreene** [comment]
The ticket seems to reference the correct tickets now, thanks @dtfranz
/unhold
@kuiwang02 would you mind reviewing and adding the `cherry-pick-approved` label if it looks good?

**Comment by kuiwang02** [comment]
@awgreene @dtfranz 
pre-verification fails. you could refer to https://issues.redhat.com/browse/OCPBUGS-18305 to get the detailed.
thanks

**Comment by kuiwang02** [comment]
> @awgreene @dtfranz pre-verification fails. you could refer to https://issues.redhat.com/browse/OCPBUGS-18305 to get the detailed. thanks

@awgreene @dtfranz after getting your information, pre-verify pass with adding sleep 5s before deleting sub and csv and adding sleep 60s before reinstalling operator.
Note: when I verify it on 4.14 and it does not need it. it is needed because of https://github.com/operator-framework/operator-lifecycle-manager/issues/3010

you could refer to https://issues.redhat.com/browse/OCPBUGS-18305 for more information.

**Pull request merged**
